File: epi_cpp/01_PrimitiveTypes/12_reverse_digits.cpp
#include <iostream>
#include <cmath>

using namespace std;

long Reverse(int x){
    bool is_negative = x<0;
    long result = 0, x_remaining = abs(x);
    while(x_remaining){
        result = (result*10) + (x_remaining%10);
        x_remaining /= 10;
    }
    return is_negative ? -result : result;
}


int main(){
    int x = -1234;
    cout << Reverse(x) << endl;
    return 0;
}

File: epi_python/01_PrimitiveTypes/12_reverse_digits.py
def reverse_digits(x: int)->int:
    result, x_remianing = 0, abs(x)
    while x_remianing:
        result = result*10 + x_remianing%10
        x_remianing //= 10
    return -result if x < 0 else result

if __name__ == '__main__':
    print(reverse_digits(-1234))